src/nanoemoji/write_part_file.py

```diff
@@ -12,37 +12,43 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Generates a part file from 1..N input part sources.
-
-Part sources can be:
-
-1. Other part files
-2. Svg files
-
-Or any mix thereof.
+"""Generates a part file from 1 source, typically an svg file.
 """
 
 
 from absl import app
 from absl import flags
-from functools import reduce
-from nanoemoji.parts import ReuseableParts
+from nanoemoji.parts import ReusableParts
 from nanoemoji import util
 from pathlib import Path
+from picosvg.geometric_types import Rect
+from picosvg.svg import SVG
 
 
 FLAGS = flags.FLAGS
 
 
+flags.DEFINE_bool("compute_donors", False, "Whether to compute donors.")
+
+
 def main(argv):
-    parts = [ReuseableParts.load(Path(part_file)) for part_file in argv[1:]]
-    if not parts:
-        raise ValueError("Specify at least one input")
-    parts = reduce(lambda a, c: a.add(c), parts[1:], parts[0])
+    if len(argv) != 2:
+        raise ValueError("Specify exactly one input")
+
+    view_box = Rect(0, 0, FLAGS.upem, FLAGS.upem)
```

Comment:
the relationship between an SVG viewBox and a font UPEM in nanoemoji is a bit more complex than that (remember how we scale to fit the viewBox.h to ascender+descender (not the UPEM proper) with the width centered or proportional).
I wonder, why did you feel you needed to add a `view_box` field to `ReusableParts` class? Why not just default to the unitary square (0, 0, 1, 1)? Doesn't reuse_tolerance already control how fine grained we want the reuse to be? What does a view_box (somewhat ambiguously assimilated to a --upem) add to this?

Reply:
ah ok, the view_box on a part file acts as a reference point for the shape paths contained therein, just like the SVG.viewBox proper; so when you add a new part or merge two part files you can scale them accordingly so they are the same relative size.
Note the viewBox can vary in width for non-emoji fonts with proportional width; I believe you call Affine2D.rect_to_rect without preserveAspectRatio so you'd see their rectangular viewbox stretched to fit the nominal UPEM square, with shapes thus distorted..
Maybe that's ok, because when it's time to _use_ the combined reusable parts, I think you will have to remap again from this ReusableParts.view_box (currently set to (0, 0, upem, upem)) to the final, actual glyph area (delimited by ascender, descender, origin and advance width).

Reply:
I want all the parts for all glyphs to exist in the same space. Using a unit square would be lovely if scaling down to it and back up again didn't cause damage.

Idea was to stick the parts into the destination coordinate system to minimize damage from repeat transform. As you point out, perhaps we should do a square with asc-desc dimensions so the transform into the destination space has no scale. In fact, I should probably assert sx and sy are almost zero.

Reply:
I changed it to use asc-desc. Mapping the shape back into font space isn't in this PR; I'll add a note to the todo list in #414.

Reply:
I'll add a test for *not* butchering shapes when loading them into the part file, that's a good catch.

Reply:
test_squares_stay_squares. Confirmed they were in fact getting mangled. I believe this is now fixed.